src/ingest/IngestSpec.cpp

```diff
@@ -468,6 +468,12 @@ bool IngestSpec::ingest(const std::string& file, BlockList& blocks) noexcept {
   // get table schema and create a table
   const auto schema = TypeSerializer::from(table_->schema);
 
+  // remove columns that are expected from macro from the schema
+  for (auto& macroCombination : macroCombinations_) {
+    const std::string colName(macroCombination.first);
+    schema->remove(colName);
+  }
+
```

Comment:
remove and add back operation looks wired to me - what about if do the treatment in line 480~483, so the `columns` will not include the partition/macro columns, and those readers like ParquetReader will not enforce on these columns missing from the file, this option seems more consistent.

Reply:
@shawncao only the csv reader looks at the columns vector, the rest use the schema directly. so that won't work

Reply:
I meant passing this `columns` to all the readers if this is a better approach than modifying schema back and forth.

Reply:
by the way. you don't need to worry about this if you don't want to include it, as this is a refactoring thing - I can take care of it later.